### ISSUE TITLE
Hyphen (-) not recognized in idnumber

### ISSUE THREAD

**Issue by monicangela** [academic-moodle-cooperation/moodle-mod_offlinequiz#23]
Hi,

My id numbers are made of letters, hyphen and numbres, for example apo-452187. When I try to change it in the params I get an error. 

For example I try this : apo-[6]=idnumber and it's not working neither on the Moodle 3 or Moodle 2.9 version of the plugin. But it works for the Moodle 2.6

Do you have any idea what's going on ?

Thanks

**Comment by univietw** [comment]
Thank you for reporting this problem.
I found the problem on the regular expression, which doesn't support hyphens at the moment, just numbers and letters, which leads to the problem you described.

I asked the former developer of this code and he explained, that we changed the way the evaluation of the id-number setting works anywhere between 2.6 and 2.9. I made a hotfix for you (see the diff attached to this comment) and will discuss, which characters should be supported, but I think there is no need to not support hyphens and some other more frequently used characters.

The change will then have to be tested by our tester and is then most certainly added to the upcoming 3.2-branch, so you won't need this hotfix any more.

I hope this fixes your problem.
[github_23_diff.txt](https://github.com/academic-moodle-cooperation/moodle-mod_offlinequiz/files/519736/github_23_diff.txt)
